test/tstQueryTreeComparisonWithBoost.cpp

```diff
@@ -160,7 +160,10 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(boost_rtree_nearest_predicate, TreeTypeTraits,
       1, std::floor(sqrt(nx * nx + ny * ny + nz * nz)));
   for (unsigned int i = 0; i < n_points; ++i)
   {
-    k_host[i] = distribution_k(generator);
+    // make sure that at least few have k = 1, as we have a separate path for
+    // this special case
+    bool const use_k1 = (i % 37);
+    k_host[i] = (!use_k1 ? distribution_k(generator) : 1);
```

Comment:
Just wanted to note that the original test also failed for incorrect implementation. But as there is no guarantee in that case, and may change on a different random number generator and/or `nx/ny/nz` parameters, I decided to add a simple check guaranteeing that for at least a few points.